case-lib/lib.sh

```diff
@@ -752,17 +752,84 @@ func_lib_check_pa()
 # However, 1. arrays would complicate the user interface 2. ALSA does not
 # seem to need arguments with whitespace or globbing characters.
 
+# SOF_ALSA_TOOL:
+# This option is used for selecting tool for testing,
+# So far, supported tools are 'alsa' and 'tinyalsa'
+# To select appropriate tool, set SOF_ALSA_TOOL to one of above
+# before using 'aplay_opts' or 'arecord_opts' function.
+# (e.g., SOF_ALSA_TOOL=alsa)
+
+
+# Function to extract the card number and device number from $dev option (e.g., hw:0,10)
+parse_audio_device() {
+    # Extract the card number (e.g., "0" from hw:0,10)
+    card_nr=$(printf '%s' "$dev" | cut -d ':' -f2 | cut -d ',' -f1)
+
+    # Extract the device number (e.g., "10" from hw:0,10)
+    dev_nr=$(printf '%s' "$dev" | cut -d ',' -f2)
+}
+
+# Function to extract the numeric format value from the $fmt_elem option
+extract_format_number() {
+    # (e.g., extracting '16' from 'S16_LE')
+    # shellcheck disable=SC2154
+    format=$(printf '%s' "$fmt_elem" | grep '[0-9]\+' -o)
+}
+
+# Initialize the parameters using for audio testing.
+initialize_audio_params()
+{
```

Comment:
Same as above
```suggestion
{
   local idx="$1"
   
```
This will temporarily shadow any global `idx` as expected.